drivers/ethernet/eth_mcux.c

```diff
@@ -1059,6 +1059,12 @@ static void eth_mcux_init(const struct device *dev)
 			kENET_RxAccelProtoCheckEnabled;
 	}
 
+	/* Set ENET2 ref clock to be generated by External OSC,*/
+	/* direction as output and frequency to 50MHz */
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(enet2), okay)
+	IOMUXC_GPR->GPR1 |= (IOMUXC_GPR_GPR1_ENET2_CLK_SEL_MASK |
```

Comment:
This looks like it should be handled at the `soc_rt10xx.c` level, not within the driver. Also, is there an `IOMUXC` API present for this bit, like `kIOMUXC_GPR_ENET1TxClkOutputDir` for ENET1?

Reply:
Thanks @danieldegrasse .. I have modified as per your recommendation.. please check now..